test/CadlRanchProjectsNonAzure.Tests/type-property-value-types.cs

```diff
@@ -0,0 +1,441 @@
+            Assert.AreEqual(new InnerEnum("UnknownValue"), response.Value.Property);
+        });
+
+        [Test]
+        public Task Type_Property_ValueTypes_ExtensibleEnum_put() => Test(async (host) =>
+        {
+            var response = await new ValueTypesClient(host, null).GetExtensibleEnumClient().PutAsync(new ExtensibleEnumProperty(new InnerEnum("UnknownValue")));
+            Assert.AreEqual(204, response.GetRawResponse().Status);
+        });
+
+        [Test]
+        public Task Type_Property_ValueTypes_Model_get() => Test(async (host) =>
+        {
+            var response = await new ValueTypesClient(host, null).GetModelClient().GetModelAsync();
+            Assert.AreEqual("hello", response.Value.Property.Property);
+        });
+
+        [Test]
+        public Task Type_Property_ValueTypes_Model_put() => Test(async (host) =>
+        {
+            var response = await new ValueTypesClient(host, null).GetModelClient().PutAsync(new ModelProperty(new InnerModel("hello")));
+            Assert.AreEqual(204, response.GetRawResponse().Status);
+        });
+
+        [Test]
+        public Task Type_Property_ValueTypes_CollectionsString_get() => Test(async (host) =>
+        {
+            var response = await new ValueTypesClient(host, null).GetCollectionsStringClient().GetCollectionsStringAsync();
+            CollectionAssert.AreEqual(new[] { "hello", "world" }, response.Value.Property);
+        });
+
+        [Test]
+        public Task Type_Property_ValueTypes_CollectionsString_put() => Test(async (host) =>
+        {
+            var response = await new ValueTypesClient(host, null).GetCollectionsStringClient().PutAsync(new CollectionsStringProperty(new[] { "hello", "world" }));
+            Assert.AreEqual(204, response.GetRawResponse().Status);
+        });
+
+        [Test]
+        public Task Type_Property_ValueTypes_CollectionsInt_get() => Test(async (host) =>
+        {
+            var response = await new ValueTypesClient(host, null).GetCollectionsIntClient().GetCollectionsIntAsync();
+            CollectionAssert.AreEqual(new[] { 1, 2 }, response.Value.Property);
+        });
+
+        [Test]
+        public Task Type_Property_ValueTypes_CollectionsInt_put() => Test(async (host) =>
+        {
+            var response = await new ValueTypesClient(host, null).GetCollectionsIntClient().PutAsync(new CollectionsIntProperty(new[] { 1, 2 }));
+            Assert.AreEqual(204, response.GetRawResponse().Status);
+        });
+
+        [Test]
+        public Task Type_Property_ValueTypes_CollectionsModel_get() => Test(async (host) =>
+        {
+            var response = await new ValueTypesClient(host, null).GetCollectionsModelClient().GetCollectionsModelAsync();
+            var result = response.Value;
+            Assert.AreEqual("hello", result.Property[0].Property);
+            Assert.AreEqual("world", result.Property[1].Property);
```

Comment:
could we assert the count of `result.Property` is `2` here as well?

Reply:
fixed